testcontainers/src/clients/http.rs

```diff
@@ -209,18 +209,43 @@ impl Http {
 
         ContainerAsync::new(container_id, client, image, self.inner.command).await
     }
-}
 
-impl Http {
     fn new() -> Self {
         Http {
             inner: Arc::new(Client {
                 command: env::command::<env::Os>().unwrap_or_default(),
-                bollard: Docker::connect_with_http_defaults().unwrap(),
+                bollard: Self::default_bollard_client(),
                 created_networks: RwLock::new(Vec::new()),
             }),
         }
     }
+}
+
+impl Http {
+    fn default_bollard_client() -> Docker {
+        use std::env;
+        use url::Url;
+
+        let host = env::var("DOCKER_HOST")
+            .ok()
+            .map(|ref h| Url::parse(h).expect(&format!("Can't parse DOCKER_HOST: {h}")));
+
+        match host {
+            Some(x) if !matches!(x.scheme(), "unix" | "npipe") => {
+                #[cfg(feature = "ssl")]
+                {
+                    Docker::connect_with_ssl_defaults()
+                }
+                #[cfg(not(feature = "ssl"))]
+                {
+                    Docker::connect_with_http_defaults()
+                }
```

Comment:
We should connect based on the presence of `DOCKER_CERT_PATH` and not, whether the feature is enabled.

Features are additive so any dependency on your dependency tree can activate it which could break, how you are connecting to the docker daemon.